templates/CRM/Contribute/Form/Task/PDFLetter.tpl

```diff
@@ -21,14 +21,16 @@
     <table class="form-layout-compressed">
       <tr><td class="label-left">{$form.thankyou_update.html} {$form.thankyou_update.label}</td><td></td></tr>
       <tr><td class="label-left">{$form.receipt_update.html} {$form.receipt_update.label}</td><td></td></tr>
-      <tr>
-        <td class="label-left">{$form.group_by.label} {help id="id-contribution-grouping"}</td>
-        <td>{$form.group_by.html}</td>
-      </tr>
-      <tr>
-        <td class="label-left">{$form.group_by_separator.label}</td>
-        <td>{$form.group_by_separator.html}</td>
-      </tr>
+      {if !$single}
+        <tr>
+          <td class="label-left">{$form.group_by.label} {help id="id-contribution-grouping"}</td>
+          <td>{$form.group_by.html}</td>
+        </tr>
```

Comment:
If you don't have this on the template then you get a different type of notice in "single" mode: `Notice: Undefined index: group_by in CRM_Contribute_Form_Task_PDFLetter->postProcess() (line 177 of .../CRM/Contribute/Form/Task/PDFLetter.php)`.

Reply:
@demeritcowboy I hit that enotice writing the test in another PR & added this line to the other PR that addresses https://github.com/civicrm/civicrm-core/pull/20188/files#diff-32b6d0d15de0f43a21f476f94999902d89e3498fc00ee55e4c817ec767a4d42fR179

Reply:
Ok.
The "other issue" referred to in the original description here I think was that an intermediate version of #20161 had more manipulating of contribution ids and it was giving buildContributionArray a null.